File: grimdb/sdk/java/src/main/java/com/grimlocker/sdk/model/FileEntry.java
package com.grimlocker.sdk.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.grimlocker.sdk.Entry;
import java.util.Map;

@JsonIgnoreProperties(ignoreUnknown = true)
public class FileEntry {

    @JsonProperty("id")                public String id;
    @JsonProperty("manifest_block_id") public String manifestBlockId;
    @JsonProperty("file_name")         public String fileName;
    @JsonProperty("mime_type")         public String mimeType;
    @JsonProperty("total_size")        public long totalSize;
    @JsonProperty("folder_id")         public String folderId;
    @JsonProperty("created_at")        public long createdAt;

    public Map<String, String> toFields() {
        return Map.of("manifest_block_id", manifestBlockId == null ? "" : manifestBlockId,
                      "file_name",         fileName == null ? "" : fileName,
                      "mime_type",         mimeType == null ? "" : mimeType,
                      "total_size",        String.valueOf(totalSize),
                      "folder_id",         folderId == null ? "" : folderId);
    }

    public static FileEntry fromEntry(Entry e) {
        FileEntry f = new FileEntry();
        if (e == null) return f;
        f.id              = e.id;
        f.manifestBlockId = e.field("manifest_block_id");
        f.fileName        = e.field("file_name");
        f.mimeType        = e.field("mime_type");
        try { f.totalSize = Long.parseLong(e.field("total_size")); } catch (NumberFormatException ignored) {}
        f.folderId        = e.field("folder_id");
        f.createdAt       = e.createdAt;
        return f;
    }

    @Override
    public String toString() {
        return "FileEntry{manifestBlockId='" + manifestBlockId + "', fileName='" + fileName + "', size=" + totalSize + "}";
    }
}

File: grimdb/sdk/csharp/src/Models/FileEntry.cs
using System.Text.Json.Serialization;

namespace Grimlocker.Models;

public sealed record FileEntry
{
    [JsonPropertyName("id")]
    public string Id { get; init; } = "";

    [JsonPropertyName("file_name")]
    public string FileName { get; init; } = "";

    [JsonPropertyName("mime_type")]
    public string MimeType { get; init; } = "";

    [JsonPropertyName("total_size")]
    public long TotalSize { get; init; }

    [JsonPropertyName("manifest_block_id")]
    public string ManifestBlockId { get; init; } = "";

    [JsonPropertyName("folder_id")]
    public string FolderId { get; init; } = "";
}
